xbmc/pictures/Picture.cpp

```diff
@@ -253,20 +253,21 @@ bool CPicture::CacheTexture(uint8_t *pixels, uint32_t width, uint32_t height, ui
     uint32_t dest_width_aligned = ((dest_width + 15) & ~0x0f);
     uint32_t stride = dest_width_aligned * sizeof(uint32_t);
 
-    uint32_t* buffer = new uint32_t[dest_width_aligned * dest_height + 4];
+    auto buffer = std::make_unique<uint32_t[]>(dest_width_aligned * dest_height + 4);
```

Comment:
Maybe outside the scope but this seems better suited to a vector?

Reply:
If I would write this from scratch I would use a vector as well. But objectively the benefit is very small: There is no resizing going on and most of the time you would bypass the vector anyway and access the raw memory with vector::data().

So in the context of the existing code I don't think it's worth to make the switch to vector.

Reply:
Yes sure, maybe an exercise for the future as it may touch quite a few interfaces.